pennylane/resources/first_quantization.py

```diff
@@ -15,505 +15,558 @@
+        3386558458840
 
-            \lambda^{\alpha}_{\nu} = \alpha \sum_{\nu \in G_0} \frac{\left \lceil
-            \mathcal{M}(2^{\mu - 2}) / \left \| \nu \right \|^2 \right \rceil}{\mathcal{M}
-            2^{2\mu - 4}},
-
-        which we compute here, for :math:`\alpha = 1`, as
+        >>> algo.qubits  # estimated number of qubits
+        3716
+    """
 
-        .. math::
+    num_wires = AnyWires
+    grad_method = None
 
-            \lambda^{1}_{\nu} = \lambda_{\nu} + \epsilon_l,
+    def __init__(
+        self,
+        n,
+        eta,
+        omega,
+        error=0.0016,
+        charge=0,
+        br=7,
+    ):
+        self.n = n
+        self.eta = eta
+        self.omega = omega
+        self.error = error
+        self.charge = charge
+        self.br = br
 
-        where :math:`\epsilon_l` is simply defined as the difference of :math:`\lambda^{1}_{\nu}`
-        and :math:`\lambda_{\nu}`. We follow Eq. (113) of
-        `PRX Quantum 2, 040332 (2021) <https://link.aps.org/doi/10.1103/PRXQuantum.2.040332>`_ to
-        derive an upper bound for its absolute value:
+        self.lamb = self.norm(self.n, self.eta, self.omega, self.error, self.br, self.charge)
 
-        .. math::
+        self.gates = self.gate_cost(self.n, self.eta, self.omega, self.error, self.br, self.charge)
+        self.qubits = self.qubit_cost(self.n, self.eta, self.omega, self.error, self.charge)
 
-            \epsilon_l = \frac{4}{2^{n_m}} (7 \times 2^{n_p + 1} + 9 n_p - 11 - 3 \times 2^{-n_p}),
+        super().__init__(wires=range(self.qubits))
 
-        where :math:`\mathcal{M} = 2^{n_m}` and :math:`n_m` is defined in Eq. (132) of
-        `PRX Quantum 2, 040332 (2021) <https://link.aps.org/doi/10.1103/PRXQuantum.2.040332>`_.
-        Finally, for :math:`p_{\nu}` defined in Eq. (128) of
-        `PRX Quantum 2, 040332 (2021) <https://link.aps.org/doi/10.1103/PRXQuantum.2.040332>`_
+    def success_prob(self, n, br):
+        r"""Return the probability of success for state preparation.
 
-        .. math::
+        The expression for computing the probability of success is taken from Eqs. (59, 60) of
+        [`PRX Quantum 2, 040332 (2021) <https://link.aps.org/doi/10.1103/PRXQuantum.2.040332>`_].
 
-            p_{\nu} = \sum_{\mu = 2}^{n_p + 1} \sum_{\nu \in B_{\mu}} \frac{\left \lceil M(2^{\mu-2}
-            / \left \| \nu \right \|)^2 \right \rceil}{M 2^{2\mu} 2^{n_{\mu} + 1}},
+        Args:
+            n (int): number of plane waves
+            br (int): number of bits for ancilla qubit rotation
 
-        we use the upper bound from Eq. (29) in
-        `arXiv:1807.09802v2 <https://arxiv.org/abs/1807.09802v2>`_ which gives
-        :math:`p_{\nu} = 0.2398`.
-    """
-    if n <= 0 or not isinstance(n, int):
-        raise ValueError("The number of planewaves must be a positive integer.")
+        Returns:
+            float: probability of success for state preparation
```

Comment:
state preparation of ... ? We should say state preparation of the rotation ancilla

Reply:
I am quoting the ref:

> The probability of success of the state preparation is usually above 0.999 for br = 7 and is

If it is not clear enough, I can use "preparation of the rotation ancilla".

Reply:
Yes, it would be better to make it clearer not here but in the upper main tagline.

Reply:
On a second thought it looks alright :)